### PR TITLE
Fix `BatchNormalization` for NHWC without cudnn

### PR DESCRIPTION
**Pull request by emcastillo**
Miss configuration in the tests made nhwc batchnorm test to use always cudnn.
There was a bug when using NHWC without cudnn due to doing reductions in the incorrect axes.

### CODE REVIEW

File: chainer/functions/normalization/batch_normalization.py [hunk 1]
```diff
@@ -426,12 +426,18 @@ def forward(self, inputs):
 
         self._impl = self._impl_selector(self, inputs)
 
+        axis = self.axis
```
Comment (niboshi) [inline chainer/functions/normalization/batch_normalization.py]: Let's use different name to avoid confusion.
```suggestion
        raw_axis = self.axis
```

**Comment by niboshi** [comment]
Please remove test skips for CPU backends.

**Comment by emcastillo** [comment]
Ideep segfaults with NHWC, I am not sure if we should add a warning.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please
